kaldb/src/main/java/com/slack/kaldb/logstore/LuceneIndexStoreConfig.java

```diff
@@ -47,26 +44,15 @@ public static Duration getRefreshDuration(final long refreshDurationSecs) {
   }
 
   public LuceneIndexStoreConfig(
-      Duration commitDuration,
-      Duration refreshDuration,
-      String indexRoot,
-      int ramBufferSizeMB,
-      boolean enableTracing) {
-    this(
-        commitDuration,
-        refreshDuration,
-        indexRoot,
-        DEFAULT_LOG_FILE_NAME,
-        ramBufferSizeMB,
-        enableTracing);
+      Duration commitDuration, Duration refreshDuration, String indexRoot, boolean enableTracing) {
+    this(commitDuration, refreshDuration, indexRoot, DEFAULT_LOG_FILE_NAME, enableTracing);
   }
 
   public LuceneIndexStoreConfig(
       Duration commitDuration,
       Duration refreshDuration,
       String indexRoot,
       String logFileName,
-      int ramBufferSizeMB,
```

Comment:
I think we would need this field exposed still. Setting the RamBufferSize controls how frequently we flush the docs to disk which in turn affects the indexing throughput, cpu usage on indexing and data freshness. At 2MBPS, we flush every ~8 secs with the default value of 16MBPS. I think we should increase the value over default so we flush every 30 seconds or so for larger segments and more efficient compression and less merge overhead. Further, we would need to tune this value depending on how much data is written to each partition by the pre-processor.

Reply:
Let's see how the defaults work before attempting to override these values, as overriding these has downstream effects on how segments are flushed. OpenSearch does override this to a percentage of the overall total ([1](https://github.com/opensearch-project/OpenSearch/blob/a0030dfb47b4312eb4087f17cc3cee0139c74b6a/server/src/main/java/org/opensearch/index/engine/InternalEngine.java#L2470), [2](https://github.com/opensearch-project/OpenSearch/blob/a0030dfb47b4312eb4087f17cc3cee0139c74b6a/server/src/main/java/org/opensearch/index/engine/EngineConfig.java#L303)), but they're intentionally doing to take advantage of a custom `IndexingMemoryController`.

If we do intend to expose this as a config this needs reworked, as this setting isn't actually settable - it's just exposed as a param in this class and then just called with hardcoded values. I'll put the ramBuffer and compound file changes into a separate PR.

Reply:
https://github.com/slackhq/kaldb/pull/313